test/test_puma_server_ssl.rb

```diff
@@ -332,6 +332,26 @@ def test_verify_client_cert
       http.verify_mode = OpenSSL::SSL::VERIFY_PEER
     end
   end
+
+  def test_verify_client_cert_using_truststore
```

Comment:
Wouldn't this test be green on master branch (minus the lines setting the truststore, of course)?

Reply:
yeah, right due the backwards compatibility when `truststore` isn't set ... 
what's confusing is that even if I set it to a `truststore` that does not contain the expected CA it still does not behave - seems like the SSL errors are not propagated properly from the Java MiniSSL impl, going to investigate.

Reply:
@kares

> SSL errors are not propagated properly from the Java MiniSSL imp

Until PR's #2847 and #2848, the MiniSSL c implementation was assumed that all cert/key/ca data was valid.  I'm pretty sure the Java implementation needs the same fixes.

Thanks for your help with JRuby in Puma...